tests/video/test_video_classifier.py

```diff
@@ -0,0 +1,135 @@
+
+if _PYTORCH_VIDEO_AVAILABLE:
+    from pytorchvideo.data.utils import thwc_to_cthw
+
+
+def create_dummy_video_frames(num_frames: int, height: int, width: int):
+    y, x = torch.meshgrid(torch.linspace(-2, 2, height), torch.linspace(-2, 2, width))
+    data = []
+    for i in range(num_frames):
+        xc = float(i) / num_frames
+        yc = 1 - float(i) / (2 * num_frames)
+        d = torch.exp(-((x - xc)**2 + (y - yc)**2) / 2) * 255
+        data.append(d.unsqueeze(2).repeat(1, 1, 3).byte())
+    return torch.stack(data, 0)
+
+
+# https://github.com/facebookresearch/pytorchvideo/blob/4feccb607d7a16933d485495f91d067f177dd8db/tests/utils.py#L33
+@contextlib.contextmanager
+def temp_encoded_video(num_frames: int, fps: int, height=10, width=10, prefix=None):
+    """
+    Creates a temporary lossless, mp4 video with synthetic content. Uses a context which
+    deletes the video after exit.
+    """
+    # Lossless options.
+    video_codec = "libx264rgb"
+    options = {"crf": "0"}
+    data = create_dummy_video_frames(num_frames, height, width)
+    with tempfile.NamedTemporaryFile(prefix=prefix, suffix=".mp4") as f:
+        f.close()
+        io.write_video(f.name, data, fps=fps, video_codec=video_codec, options=options)
+        yield f.name, thwc_to_cthw(data).to(torch.float32)
+    os.unlink(f.name)
+
+
+@contextlib.contextmanager
+def mock_encoded_video_dataset_file():
+    """
+    Creates a temporary mock encoded video dataset with 4 videos labeled from 0 - 4.
+    Returns a labeled video file which points to this mock encoded video dataset, the
+    ordered label and videos tuples and the video duration in seconds.
+    """
+    num_frames = 10
+    fps = 5
+    with temp_encoded_video(num_frames=num_frames, fps=fps) as (
+        video_file_name_1,
+        data_1,
+    ):
+        with temp_encoded_video(num_frames=num_frames, fps=fps) as (
+            video_file_name_2,
+            data_2,
+        ):
+            with tempfile.NamedTemporaryFile(delete=False, suffix=".txt") as f:
+                f.write(f"{video_file_name_1} 0\n".encode())
+                f.write(f"{video_file_name_2} 1\n".encode())
+                f.write(f"{video_file_name_1} 2\n".encode())
+                f.write(f"{video_file_name_2} 3\n".encode())
+
+            label_videos = [
+                (0, data_1),
+                (1, data_2),
+                (2, data_1),
+                (3, data_2),
+            ]
+            video_duration = num_frames / fps
+            yield f.name, label_videos, video_duration
+
+
+@pytest.mark.skipif(not _PYTORCH_VIDEO_AVAILABLE, reason="PyTorch Video isn't installed.")
+def test_image_classifier_finetune(tmpdir):
+
+    _EPS = 1e-9
+
+    with mock_encoded_video_dataset_file() as (
+        mock_csv,
+        label_videos,
+        total_duration,
+    ):
+
+        half_duration = total_duration / 2 - _EPS
+
+        datamodule = VideoClassificationData.from_folders(
+            train_folder=mock_csv,
```

Comment:
So it's surprising to me that you're using the from_folders method but passing in a csv. You should either pass in a folder structured input, or change the name of the function to be generic e.g. from_path

Reply:
Yes, I will align our API with yours, but I need to do so with ImageClassifer to.
I propose `from_paths`

Reply:
For ImageClassifier, we have `from_filepaths` which matches "path_name, "label".

Reply:
Created a ticket from myself: https://github.com/PyTorchLightning/lightning-flash/issues/218